File: simter-id-data/src/main/java/tech/simter/id/po/IdHolder.java
package tech.simter.id.po;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import java.io.Serializable;

/**
 * ID holder.
 *
 * @author RJ
 */
@Entity
public class IdHolder implements Serializable {
  /**
   * The type.
   */
  @Id
  @Column(nullable = false, length = 100)
  public String type;

  /**
   * The current value.
   */
  @Column(nullable = false)
  public Long value;

  public IdHolder() {
  }

  public IdHolder(String type, Long value) {
    this.type = type;
    this.value = value;
  }
}

File: simter-id-data/src/main/kotlin/tech/simter/id/po/IdHolder.kt
package tech.simter.id.po

import tech.simter.id.TABLE_ID
import javax.persistence.Column

/**
 * The ID holder.
 *
 * @author RJ
 */
// for jpa
@javax.persistence.Entity
@javax.persistence.Table(name = TABLE_ID)
// for jdbc or r2dbc
@org.springframework.data.relational.core.mapping.Table(TABLE_ID)
// for mongodb
@org.springframework.data.mongodb.core.mapping.Document(collection = TABLE_ID)
data class IdHolder(
  /** The type */
  @javax.persistence.Id
  @org.springframework.data.annotation.Id
  @Column(length = 100)
  val t: String,
  /** The current value */
  val v: Long
)